soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -323,7 +334,7 @@ __weak void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 			imr_layout->imr_state.header.imr_ram_storage = NULL;
 		}
 #endif /* CONFIG_ADSP_IMR_CONTEXT_SAVE */
-
+		z_mark_thread_as_not_suspended(z_current_get());
```

Comment:
This isn't right.  The "suspended" state means that a thread is neither running currently nor in the run queue[1].  By definition, an idle thread can never be blocked, and by implementation choice is never added to the run queue.  It just doesn't make sense here.